sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/labmessage/processing/AbstractLabMessageProcessingFlow.java

```diff
@@ -254,38 +225,18 @@ private CompletionStage<ProcessingResult<SampleAndPathogenTests>> handleRelatedL
 		});
 	}
 
-	private CompletionStage<ProcessingResult<PersonDto>> pickOrCreatePerson(ExternalMessageDto labMessage) {
-
-		final PersonDto person = buildPerson(ExternalMessageMapper.forLabMessage(labMessage));
-
-		HandlerCallback<PersonDto> callback = new HandlerCallback<>();
-		handlePickOrCreatePerson(person, callback);
-
-		return callback.futureResult;
-	}
-
-	protected abstract void handlePickOrCreatePerson(PersonDto person, HandlerCallback<PersonDto> callback);
-
-	private PersonDto buildPerson(ExternalMessageMapper mapper) {
-
-		final PersonDto personDto = PersonDto.build();
-
-		mapper.mapToPerson(personDto);
-		mapper.mapToLocation(personDto.getAddress());
-
-		return personDto;
-	}
-
-	private CompletionStage<ProcessingResult<PersonAndPickOrCreateEntryResult>> pickOrCreateEntry(PersonDto person, ExternalMessageDto labMessage) {
+	private CompletionStage<ProcessingResult<PersonAndPickOrCreateEntryResult>> pickOrCreateEntry(
+		PersonDto person,
+		ExternalMessageDto externalMessage) {
 
 		PersonReferenceDto personRef = person.toReference();
-		List<CaseSelectionDto> similarCases = getSimilarCases(personRef, labMessage);
-		List<SimilarContactDto> similarContacts = getSimilarContacts(personRef, labMessage);
-		List<SimilarEventParticipantDto> similarEventParticipants = getSimilarEventParticipants(labMessage, personRef);
+		List<CaseSelectionDto> similarCases = getSimilarCases(personRef, externalMessage);
+		List<SimilarContactDto> similarContacts = getSimilarContacts(personRef, externalMessage);
+		List<SimilarEventParticipantDto> similarEventParticipants = getSimilarEventParticipants(externalMessage, personRef);
```

Comment:
Would be nice to use the same order for the parameters as the methods above. so 1st have the `personRef`, then the `externalMessage`